Word Search Problem

```diff
@@ -15,10 +15,10 @@ class Solution:
                 
     def backtrack(self,board,word,i,jindex) -> bool:
```

Comment:
class Solution:
    def __init__(self):
        self.m=None
        self.n=None
    def exist(self, board: List[List[str]], word: str) -> bool:        
        self.m=len(board)
        self.n=len(board[0])       
       
        for i in range (self.m):
            for j in range (self.n):
                if (self.backtrack(board, word, i, j, 0)):
                    return True
        return False
    def backtrack(self, board, word, i, j, index):
        ### Base Case
        if index == len(word):
            return True 
        
        if i < 0 or j < 0 or i >= self.m or j >= self.n or board[i][j] == '#': 
            return False
           
        ### Logic
        dirs = [(0,1),(1,0),(-1,0),(0,-1)]
        
        if word[index] == board[i][j]:
            temp = board[i][j]   ### Taking a copy of the position 
            ### Action
            board[i][j] = '#'             
            ### Recursion
            for dire in dirs :
                r = dire[0] + i
                c = dire[1] + j        
                if (self.backtrack(board, word, r, c, index+1)):
                    return True
            #### BackTrack
            board[i][j]=temp
        return False